kairos-cli/src/commands/deposit.rs

```diff
@@ -15,14 +20,31 @@ pub struct Args {
     private_key_path: PrivateKeyPathArg,
 }
 
-pub fn run(args: Args) -> Result<String, CliError> {
-    let _amount: u64 = args.amount.field;
-    let _signer =
+pub fn run(args: Args, kairos_server_address: Url) -> Result<String, CliError> {
+    let amount: u64 = args.amount.field;
+    let signer =
         Signer::from_private_key_file(args.private_key_path.field).map_err(CryptoError::from)?;
 
-    // TODO: Create transaction and sign it with `signer`.
+    let client = reqwest::Client::new();
```

Comment:
Similarily to https://github.com/cspr-rad/kairos/pull/91/files#r1590918696, I would push `reqwest` client construction down to client implementation at `kairos-cli/src/client.rs`.

Reply:
[31b4c71](https://github.com/cspr-rad/kairos/pull/91/commits/31b4c715ca55533de1d583e8ad434c55f397b78c)